docs/development/release.md

```diff
@@ -3,148 +3,297 @@
 
 ## Prerequisite
 
-You need [KubeRay GitHub](https://github.com/ray-project/kuberay) write permissions to cut a release branch and create a release tag.
+You need write access to the [KubeRay repo](https://github.com/ray-project/kuberay) to cut a release branch and create a release tag.
 
-### Overview
+## Overview
 
-Each major release (e.g. `0.4`) is managed in its own GitHub branch.
-To release KubeRay, cut a release branch (e.g. `release-0.4`) from master and build commits on that branch
-until you reach a satisfactory final release commit.
+Each KubeRay minor release series (e.g., `v1.3.X`) is maintained on its own release branch. For example, the `v1.3.X` releases are cut from the `release-1.3` branch. Patch releases (e.g., `v1.3.1`) involve cherry-picking commits onto the corresponding release branch.
 
-Immediately after cutting the release branch, create a commit for a release candidate (e.g. `0.4.0-rc.0`),
-and build the associated artifacts (images and charts).
-If further changes need to be made to the release, pick changes from the master branch into the release branch.
-Make as many release candidates as necessary until a stable final release commit is reached.
-Then build final release artifacts, publish release notes, and announce the release.
+The high-level steps for a new minor or patch release are:
 
-### KubeRay release schedule
+1. For new minor releases, create a release branch (e.g., `release-1.4`) from `master` if it doesn't exist.
+2. Update KubeRay version references in the release branch.
+3. Create and push a new git tag based on the target release version (e.g., `v1.4.0`).
+4. Publish container images by running the `release-image-build` GitHub Actions workflow.
+5. Publish kubectl plugin binaries by running the `release-kubectl-plugin` GitHub Actions workflow.
+6. Update the `kuberay-helm` repository to publish the new Helm chart versions.
+7. Validate the release artifacts (images, charts).
+8. Generate the CHANGELOG for the release.
+9. Publish the release notes on GitHub Releases.
+10. Update Ray documentation to refer to the new KubeRay version.
```

Comment:
Add a step to announce the new release in Slack.

Reply:
done

Reply:
I added as a step inside 9)